### PR TITLE
feat: Backend CRUD for Modules details

### PR DESCRIPTION
**Pull request by DancinParrot**

### CODE REVIEW

**Review by DancinParrot — COMMENTED** [review]
Refactor required.

File: src/main/java/com/cloudlabs/server/Modules/ModuleController.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        // Create a new Module instance
+        Module newModule = new Module();
+
+        // Set the attributes of the module
+        newModule.setModuleSubtitle(subtitle);
+        newModule.setModuleName(title);
+        newModule.setModuleDescription(description);
+
+        // Add the new module to the modules repository
+        this.repository.save(newModule);
```
Comment (DancinParrot) [inline src/main/java/com/cloudlabs/server/Modules/ModuleController.java:97]: Use a contructor

File: src/main/java/com/cloudlabs/server/Modules/Module.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        this.moduleDescription = moduleDescription;
+    }
+    
+    public static Module from(String moduleSubtitle, String moduleName, String moduleDescription) {
+        Module module = new Module();
+        module.setModuleSubtitle(moduleSubtitle);
+        module.setModuleName(moduleName);
+        module.setModuleDescription(moduleDescription);
+
+        module.setModuleId(Long.valueOf(UUID.randomUUID().toString()));
```
Comment (DancinParrot) [inline src/main/java/com/cloudlabs/server/Modules/Module.java:66]: ID will be automatically generated by JPA when you save() an entity to the database, no need to manually generate a UUID,

File: src/main/java/com/cloudlabs/server/Modules/Module.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+        return module;
+    }
```
Comment (DancinParrot) [inline src/main/java/com/cloudlabs/server/Modules/Module.java:69]: Use a constructor

File: src/main/java/com/cloudlabs/server/Modules/ModuleController.java [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        // Convert moduleId to Long
+        Long moduleIdAsLong;
+        try {
+            moduleIdAsLong = Long.valueOf(moduleId);
+        } catch (NumberFormatException e) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Invalid module ID format");
+        }
+
+        // Find Module by its id
+        Module module = this.repository.findByModuleId(moduleIdAsLong);
```
Comment (DancinParrot) [inline src/main/java/com/cloudlabs/server/Modules/ModuleController.java:66]: Business logic and entities should not be inside the controller. Do seperate the logic into a different file, i.e. ModuleService and ModuleServiceImpl, you may refer to #19.

File: src/main/java/com/cloudlabs/server/Modules/ModuleController.java [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+        // Exception if module does not exist
+        if (module == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Module not found");
+        }
+
+        // Return the module
+        return module;
```
Comment (DancinParrot) [inline src/main/java/com/cloudlabs/server/Modules/ModuleController.java:74]: Controller should return DTO instead of entity

**Review by DancinParrot — COMMENTED** [review]
Refactor required

**Review by s10219129 — APPROVED** [review]

**Comment by DancinParrot** [comment]
Resolving conflict has caused staging (with Auth) being merged into the branch, resulting in failed tests due to no Auth being incorporated into the test.

Fix should be relatively straightforward:
```Java
@WithMockUser(username = "something", roles = { "something" })
void createModule_whenValidParametersGiven() throws Exception {
//...
}
```

**Pull request merged**